tests/base/parameterframe/test_parameterframe.py

```diff
@@ -31,6 +36,10 @@ class TestParameterFrame:
     def setup_method(self):
         self.frame = BasicFrame.from_dict(deepcopy(FRAME_DATA))
 
+    def test_frame_defined_with_dataclass_annotation(self):
+        with pytest.raises(AttributeError):
```

Comment:
this should raise a type error not an attribute error if its hitting the right error no?

Reply:
I think you want to init the frame directly, not through the constructor

Reply:
also this test flags up that we need to protect all the constructors because __dataclass_fields__ wont exist (hence the AttributeError)

Reply:
Oh hec, that's true. I initally did it with TypeError and then just thought I was wrong and changed just copied the AttributeError.